pkg/render/intrusion_detection.go

```diff
@@ -136,9 +153,31 @@ func (c *intrusionDetectionComponent) Objects() ([]client.Object, []client.Objec
 		c.intrusionDetectionRoleBinding(),
 		c.intrusionDetectionDeployment(),
 	)
+
 	objs = append(objs, secret.ToRuntimeObjects(secret.CopyToNamespace(IntrusionDetectionNamespace, c.cfg.ESSecrets...)...)...)
 	objs = append(objs, c.globalAlertTemplates()...)
-	objs = append(objs, c.intrusionDetectionADJobsPodTemplate()...)
+
+	// AD Related deployment only for management/standalone cluster
+	if !c.cfg.ManagedCluster {
```

Comment:
I think this file is getting large enough that I wonder if the AD objects should be refactored to a separate file. But I am okay to do that in a separate PR after getting this merged if the maintainers are okay with it for now.